yt_dlp/extractor/bbc.py

```diff
@@ -1255,6 +1268,48 @@ def extract_all(pattern):
                 lambda s: self._parse_json(s, playlist_id, fatal=False),
                 re.findall(pattern, webpage))))
 
+        def parse_model(model):
+            '''Extract single video from model structure'''
+            if isinstance(model, list):
+                model = model[0]
```

Comment:
I think you're getting a list when you expected a dict because the traversal is branching.
```suggestion
```